format/spec.md

```diff
@@ -218,6 +218,8 @@ Valid type promotions are:
 * `int` to `long`
 * `float` to `double`
 * `decimal(P, S)` to `decimal(P', S)` if `P' > P` -- widen the precision of decimal types.
+* `bytes` to `string`
+* `string` to `bytes`
```

Comment:
We're going to need to ensure this is implemented in Java, and add a note about compatibility. This isn't a forward-compatible change.